### PR TITLE
use the same glibc version as the docker image for cross

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by isuruf** [comment]
@conda-forge/core, this is ready for review

**Comment by carterbox** [comment]
Thanks, @isuruf! I just spent most of a day coming to the same conclusion. 😄 

Closes https://github.com/conda-forge/pytorch-cpu-feedstock/pull/299
Closes https://github.com/conda-forge/libcufile-feedstock/issues/26
Closes https://github.com/conda-forge/libcufile-feedstock/pull/27

**Review by beckermr — APPROVED** [review]
Do we still need a default somewhere or are we confident we'll always be able to find the `__glibc` virtual package?

**Comment by h-vetinari** [comment]
Thank you for this, and for adding an explanatory comment! 🙏

File: recipe/cross_compile_support.sh [hunk 1]
```diff
@@ -5,12 +5,21 @@ BUILD_PLATFORM=$(conda info --json | jq -r .platform)
...
-    GLIBC_VERSION=$(cat ${CI_SUPPORT}/${CONFIG}.yaml | shyaml get-value c_stdlib_version.0 2.17)
 fi
 
+# When compiling natively we use the docker image's glibc
+# We do the same now when cross-compiling by creating a
+# conda environment for host platform with the same glibc
+# as the docker glibc.
+# For eg: when cross compiling for linux-aarch64 in a
+# docker image with linux-anvil-x86_64:alma9 image, we use
+# glibc=2.28 for QEMU while still building the package for
```
Comment (h-vetinari) [inline recipe/cross_compile_support.sh:16]: I think the comment is slightly off - alma9 would yield a glibc of 2.34 through `__glibc`; 2.28 goes with alma8.

**Pull request merged**